Src/Base/AMReX_DistributionMapping.cpp

```diff
@@ -1410,6 +1410,183 @@ DistributionMapping::makeKnapSack (const Vector<Real>& rcost, Real& eff, int nma
 
     return r;
 }
+
+DistributionMapping
+DistributionMapping::makeKnapSack (const LayoutData<Real>& rcost_local,
+                                   Real& currentEfficiency, Real& proposedEfficiency,
+                                   int nmax, bool broadcastToAll, int root)
+{
+    BL_PROFILE("makeKnapSack");
+
+    // Proposed distribution mapping is computed from global vector of costs on root;
+    // required information is gathered on root from the layoutData information
+    //
+    // Two main steps:
+    // 1. collect from rcost_local into the global cost vector rcost; then rcost is
+    //    complete (only) on root
+    // 2. (optional; default true) Broadcast processor map of the new dm to others
+    DistributionMapping r;
+    Vector<Real> rcost;
+    rcost.resize(rcost_local.size());
+    
+#ifdef BL_USE_MPI
+    // Gather prelims
+    //
+    // Gather costs and global indices in the BoxArray to root:
+    // These gathers all have the pattern
+    //     gatherv(&my_quantity[0], my_quantity.size(),
+    //             &root_recvbuf[0], root_recvcount,
+    //             root)
+    amrex::Vector<int> index_to_send = rcost_local.IndexArray();
+    amrex::Vector<Real> cost_to_send;
+    cost_to_send.resize(rcost_local.local_size());
+    
+    for (auto i : rcost_local.IndexArray())
+    {   
+        cost_to_send[rcost_local.localindex(i)] = rcost_local[i];
+    }
+
+    amrex::Vector<long> recvcount, disp;
```

Comment:
Can we use `Long` instead?  It's an alias to `long` on Unix and `long long` on Windows.

Reply:
Long works, switched to that.